app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt

```diff
@@ -141,16 +160,24 @@ class ThreadFragment : Fragment() {
     }
 
     override fun onDestroyView() {
+        removeGlobalLayoutListener()
         threadAdapter.resetCallbacks()
         super.onDestroyView()
         _binding = null
     }
 
+    private fun listenToGlobalLayoutChanges() {
+        binding.toolbar.viewTreeObserver.addOnGlobalLayoutListener(globalLayoutListener)
+    }
+
+    private fun removeGlobalLayoutListener() {
+        binding.toolbar.viewTreeObserver.removeOnGlobalLayoutListener(globalLayoutListener)
+    }
+
     private fun setupUi() = with(binding) {
 
         toolbar.setNavigationOnClickListener {
-            val onlyThreadIsShown = (parentFragment as TwoPaneFragment).isOnlyRightShown()
-            if (onlyThreadIsShown) mainViewModel.closeThread()
+            if (twoPaneFragment.isOnlyRightShown()) mainViewModel.closeThread()
```

Comment:
We need to always close the thread when the button is clicked and only display the button when the thread can be closed.

I don't think we need bother with this logic here

Reply:
Done here: https://github.com/Infomaniak/android-kMail/pull/1592